mods/persistence/modules/modular_computers/hardware/battery_module.dm

```diff
@@ -1,10 +1,21 @@
-/obj/item/stock_parts/computer/battery_module/Initialize()
-	// Painful override to stop attempting to create a type rather than a path on load.
-	var/old_battery
+/obj/item/stock_parts/computer/battery_module
+	var/obj/item/cell/old_battery
+
+/obj/item/stock_parts/computer/battery_module/before_save()
+	. = ..()
 	if(istype(battery))
 		old_battery = battery
-		battery = /obj/item/cell/
+		battery = old_battery.type
+		CUSTOM_SV("charge", old_battery.charge)
```

Comment:
What are you doing all that for though? Why aren't you just relying on the saved cell handling its own saving?

Reply:
The saved cell is overwritten on Initialize, so the previous system would allow a new cell to be created, and then delete it afterwards. This would cause an error since a non-Initialized atom was deleted though, and in general allowing something to be created just to delete it is pretty undesirable.

Reply:
That's fine. Its just I wonder why you made a random var "var/obj/item/cell/old_battery" when you can just skip on saving the battery var and save the battery path in a CUSTOM_SV and set it on init. Its just a bit weird to have a var in the class that's just kinda there doing nothing.

Reply:
I find it preferable, since we don't need to do three steps for the deserialization, and normal vars are easier to read in the database

Reply:
Yeah, but you're making use of CUSTOM_SV anyways already. So the small overhead is already there. Plus, more vars, also implicitly makes the object's vars list slower to access. 

I'll approve the PR, but not too fond of that. That used to plague the old codebase and generally being really nasty.